src/engine/layoutstore.ts

```diff
@@ -133,7 +137,8 @@ class LayoutStore {
         delete this.store[key_without_activity];
       } else {
         let output_name = key.slice(0, key.indexOf("@"));
-        this.store[key] = new LayoutStoreEntry(output_name);
+        let desktop_name = i2 !== -1 ? key.slice(i2 + 1) : undefined;
```

Comment:
cannot be undefined. desktop_name as split of HDMI-A-1@f381c9cf-cb90-4ade-8b3f-24ae0002d366#Desktop 1

Reply:
Name generated in [KWinSurface.generateId](https://github.com/anametologin/krohnkite/blob/master/src/driver/kwin/kwinsurface.ts#L22) and `#Desktop 1` part may be missing.

It writes as surface ID [here](https://github.com/anametologin/krohnkite/blob/master/src/driver/kwin/kwinsurface.ts#L49) and passed to [LayoutStore.getEntry](https://github.com/anametologin/krohnkite/blob/master/src/engine/layoutstore.ts#L123) ([1](https://github.com/anametologin/krohnkite/blob/master/src/engine/layoutstore.ts#L110), [2](https://github.com/anametologin/krohnkite/blob/master/src/engine/layoutstore.ts#L115), [3](https://github.com/anametologin/krohnkite/blob/master/src/engine/layoutstore.ts#L120)).